File: backend/src/main/java/com/example/backend/service/MedicationService.java
package com.example.backend.service;

import com.example.backend.dto.MedicationDto;
import com.example.backend.entity.Medication;

import java.util.List;

public interface MedicationService {
    MedicationDto saveMedication(MedicationDto medicationDto);
    List<MedicationDto> getAllMedication();
    MedicationDto getMedicationById(Long id);
    MedicationDto updateMedication(Long id, MedicationDto medicationDto);
    void deleteMedication(Long id);
    List<Medication> findMedicationByAppointmentId(Long app_id);
}

File: frontend/src/service/MedicationService.ts
import axios from "axios";
import { getToken } from "./AuthService";

const AUTH_REST_API_URL = "http://localhost:8080/api/medication";

axios.interceptors.request.use(
  function (config) {
    config.headers["Authorization"] = getToken();
    return config;
  },
  function (error) {
    return Promise.reject(error);
  }
);

export const saveMedication = (registerObj: any) =>
    axios.post(AUTH_REST_API_URL, registerObj);

export const getAllMedication = () =>
    axios.get(AUTH_REST_API_URL);

export const getMedicationById = (id:number) =>
  axios.get(AUTH_REST_API_URL + "/" + id);

export const updateMedication = (id: number, data:{}) =>
    axios.put(AUTH_REST_API_URL + "/update/" + id, data);

export const deleteMedication = (id: number) =>
  axios.delete(AUTH_REST_API_URL + "/delete" + id);

export const findMedicationByAppointmentId = (id:number) =>
  axios.get(AUTH_REST_API_URL + "/appointment/" + id);